packages/framework/fluid-static/src/types.ts

```diff
@@ -10,6 +10,12 @@ import {
 } from "@fluidframework/core-interfaces";
 import { type ISharedObjectKind } from "@fluidframework/shared-object-base";
 
+/**
+ * Valid compat modes that may be specified when creating a DOProviderContainerRuntimeFactory.
+ * @public
+ */
+export type CompatMode = "1" | "2";
```

Comment:
Is the intent to hide away details about the differences between the versions? If so, the type union makes sense to me. If not, an enum-like object would give us a place to document implications of each version.

Reply:
That pattern would also discourage the use of magic constants in consuming code.

Reply:
Yes for hiding the details - this will ultimately be paired with migration guidance documentation so providing a simple and straightforward set of instructions is a high-order goal.  I'll experiment with other options here though (probably pending getting some of that guidance written so we can see how it looks).